test/test-tools.go

```diff
@@ -157,6 +157,51 @@ func AssertNotContains(t *testing.T, haystack string, needle string) {
 	}
 }
 
+// AssertMetricEquals determines whether the value held by a prometheus Collector
+// (e.g. Gauge, Counter, CounterVec, etc) is equal to the expected integer.
+// In the case of vector collectors, it collects and sums the values from all
+// metrics in the vector prior to comparison; this is so that users can easily
+// make assertions about one of many field dimensions (e.g. for a CounterVec with
+// fields "host" and "valid", being able to assert that two "valid": "true"
+// incremets occurred, without caring which host did each increment). To make
+// assertions about subsets of a vector's metrics, use the MetricVec's .CurryWith
+// or .GetMetricWith methods. Only works for integer metrics (Counters and Gauges),
+// does not work for Histograms.
+func AssertMetricWithLabelsEquals(t *testing.T, c prometheus.Collector, l prometheus.Labels, expected int) {
+	ch := make(chan prometheus.Metric)
+	done := make(chan struct{})
+	go func() {
+		c.Collect(ch)
+		close(done)
+	}()
+	var total int
+	timeout := time.After(time.Second)
+loop:
+	for {
+	metric:
+		select {
+		case <-timeout:
+			t.Fatal("timed out collecting metrics")
+		case <-done:
+			break loop
+		case m := <-ch:
+			var iom io_prometheus_client.Metric
+			_ = m.Write(&iom)
+			for _, lp := range iom.Label {
+				val, ok := l[lp.GetName()]
+				if ok && lp.GetValue() != val {
+					break metric
+				}
+			}
+			// Exactly one of the Counter or Gauge values will be populated by the
+			// .Write() operation, so just add both because the other will be 0.
+			total += int(iom.Counter.GetValue())
+			total += int(iom.Gauge.GetValue())
```

Comment:
Fair enough. I did it this way because we only ever compare against ints in our tests, but since those are all literals I think changing the signature here to take a float64 and removing these conversions should work seamlessly.